src/views/UserResourceDetails.vue

```diff
@@ -0,0 +1,212 @@
+      </v-form>
+    </v-sheet>
+  </v-container>
+</template>
+
+<script>
+import { ref, computed, inject } from "vue";
+import { useRoute, useRouter } from "vue-router";
+import { useStore } from "vuex";
+import { useNumberFieldRules } from "../utils/validation-rules";
+import ResourceCard from "@/components/Card/ResourceCard.vue";
+
+export default {
+  components: {
+    ResourceCard,
+  },
+  props: {
+    id: String,
+  },
+  async setup(props) {
+    const form = ref(null);
+    const numberFieldRules = useNumberFieldRules();
+    const store = useStore();
+    const route = useRoute();
+    const router = useRouter();
+    const pageTitle = ref(route.meta.title);
+    const showSnackbar = inject("showSnackbar");
+    const quantity = ref("");
+    const resourceId = props.id;
+    const resouceByUser = computed(
+      () => store.getters["users/getUserResources"]
+    ); // owner + resourcesAndQuantities
+    const allUsers = computed(() => store.getters["users/allUsers"]);
+    const selectedUser = ref("");
+    const userOptions = computed(() =>
+      store.getters["users/allUsers"].map((user) => user.name)
+    );
+    const currentQuantity = ref("");
+
+    const quantityAndResourceByUser = ref([]);
+
+    for (const index in allUsers.value) {
+      const userId = allUsers.value[index].id;
+      const res = await store.dispatch("users/fetchResourcesByUser", userId);
+      quantityAndResourceByUser.value.push(res);
+    }
+
+    function getCurrentResourceQuantity() {
+      const currentResources = resouceByUser.value.resourcesAndQuantities;
+
+      for (const index in currentResources) {
+        currentResources[index].resource.id === resourceId
+          ? (currentQuantity.value += currentResources[index].quantity)
+          : "";
+      }
+    }
+
+    async function getCurrentResourceQuantityAndUser() {
+      const url = router.options.history.state.back; // /users/userId
+      const userId = url.substring(url.lastIndexOf("/") + 1); // substing to find userId
+      try {
+        await store.dispatch("users/fetchResourcesPerUser", userId);
+        const user = computed(() => store.getters["users/getUserResources"]);
+        const currentUserQuantity = user.value.resourcesAndQuantities;
+        selectedUser.value = user.value.owner.name;
+
+        for (const index in currentUserQuantity) {
+          const currentUserResource = currentUserQuantity[index].resource;
+          if (currentUserResource.id === resourceId) {
+            const resourceQuantity = currentUserQuantity[index].quantity;
+            quantity.value = resourceQuantity;
+            currentQuantity.value = resourceQuantity;
+          }
+        }
+      } catch (error) {
+        showErrorSnackbar("Couldn't get user and its resource/quantity");
+      }
+    }
+
+    if (route.path.includes("/add")) {
+      getCurrentResourceQuantity();
+    } else if (route.path.includes("/remove")) {
+      getCurrentResourceQuantityAndUser();
```

Comment:
Why do we need to differentiate between the 2 cases? In both cases we are fetching all the users with the resources they own. I think we should get rid of `getCurrentResourceQuantityAndUser`

Reply:
getCurrentResourceQuantityAndUser() - im getting the user id that im coming from so i can find the quantity for this exact resource for that user